fastly/block_fastly_service_v1_logging_logshuttle_test.go

```diff
@@ -0,0 +1,235 @@
+	name := fmt.Sprintf("tf-test-%s", acctest.RandString(10))
+	domain := fmt.Sprintf("fastly-test.%s.com", name)
+
+	log1 := gofastly.Logshuttle{
+		Version:       1,
+		Name:          "logshuttle-endpoint",
+		Token:         "s3cr3t",
+		URL:           "https://example.com",
+		FormatVersion: 2,
+		Format:        "%h %l %u %t \"%r\" %>s %b",
+	}
+
+	log1_after_update := gofastly.Logshuttle{
+		Version:       1,
+		Name:          "logshuttle-endpoint",
+		Token:         "secret",
+		URL:           "https://new.example.com",
+		FormatVersion: 2,
+		Format:        "%h %l %u %t \"%r\" %>s %b %T",
+	}
+
+	log2 := gofastly.Logshuttle{
+		Version:           1,
+		Name:              "another-logshuttle-endpoint",
+		Token:             "another-token",
+		URL:               "https://another.example.com",
+		Placement:         "none",
+		ResponseCondition: "response_condition_test",
+		FormatVersion:     2,
+		Format:            "%h %l %u %t \"%r\" %>s %b",
+	}
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckServiceV1Destroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccServiceV1LogshuttleConfig(name, domain),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckServiceV1Exists("fastly_service_v1.foo", &service),
+					testAccCheckFastlyServiceV1LogshuttleAttributes(&service, []*gofastly.Logshuttle{&log1}),
+					resource.TestCheckResourceAttr(
+						"fastly_service_v1.foo", "name", name),
+					resource.TestCheckResourceAttr(
+						"fastly_service_v1.foo", "logging_logshuttle.#", "1"),
+				),
+			},
+
+			{
+				Config: testAccServiceV1LogshuttleConfig_update(name, domain),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckServiceV1Exists("fastly_service_v1.foo", &service),
+					testAccCheckFastlyServiceV1LogshuttleAttributes(&service, []*gofastly.Logshuttle{&log1_after_update, &log2}),
+					resource.TestCheckResourceAttr(
+						"fastly_service_v1.foo", "name", name),
+					resource.TestCheckResourceAttr(
+						"fastly_service_v1.foo", "logging_logshuttle.#", "2"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckFastlyServiceV1LogshuttleAttributes(service *gofastly.ServiceDetail, logshuttle []*gofastly.Logshuttle) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+
+		conn := testAccProvider.Meta().(*FastlyClient).conn
+		logshuttleList, err := conn.ListLogshuttles(&gofastly.ListLogshuttlesInput{
+			Service: service.ID,
+			Version: service.ActiveVersion.Number,
+		})
+
+		if err != nil {
+			return fmt.Errorf("[ERR] Error looking up Log Shuttle Logging for (%s), version (%d): %s", service.Name, service.ActiveVersion.Number, err)
+		}
+
+		if len(logshuttleList) != len(logshuttle) {
+			return fmt.Errorf("Log Shuttle List count mismatch, expected (%d), got (%d)", len(logshuttle), len(logshuttleList))
+		}
+
+		log.Printf("[DEBUG] logshuttleList = %#v\n", logshuttleList)
```

Comment:
Is this still needed?

Reply:
I think it will still be useful in debugging global state.